src/Perpetuum/Services/Channels/ChatCommands/AdminCommandHandlers.cs

```diff
@@ -170,13 +170,21 @@ public static void ShutdownCancel(AdminCommandData data)
         [ChatCommand("JumpTo")]
         public static void JumpTo(AdminCommandData data)
         {
-            bool err = false; //TODO this only throws on bad last-arg!
-            err = !int.TryParse(data.Command.Args[0], out int zone);
-            err = !int.TryParse(data.Command.Args[1], out int x);
-            err = !int.TryParse(data.Command.Args[2], out int y);
-            if (err)
+            int zone;
+            int x;
+            int y;
+
+            try
             {
-                throw PerpetuumException.Create(ErrorCodes.RequiredArgumentIsNotSpecified);
+                zone = int.Parse(data.Command.Args[0]);
+                x = int.Parse(data.Command.Args[1]);
+                y = int.Parse(data.Command.Args[2]);
```

Comment:
Aha! another thing to check first (or catch and report on) is the expected argument length!
For example here, if only the zoneid is passed, or just 2 params (ints) this will have an out of bounds array access, which throws whatever the error is, but isn't informative as to the issue.

That said, it is better than it was before - but another exception type check could handle this nicely